Add filter to turn a string into a UUID

gaqzi opened the pull request:
This filter was made because I needed to create idempotent UUIDs when
installing the agent for Go (http://go.cd), which uses UUIds to
distinguish the agents from each other.

The filter doesn't do anything special when the string is empty, would this be a good idea to add an assert for?

I hope this would be useful for inclusion in core. :)

sivel commented:
I'd recommend a few changes. Such as not naming the function or the filter `uuid`, but perhaps name it something more specific to its function since the uuid provided functions all do something different and the namespace you chose is specifically for FQDN hostnames.

gaqzi commented:
@sivel: What do you think of me doing this:

1) Create an Ansible namespace and use that (it's just a UUID anyway)
2) Rename `uuid` to `uuid_from_string` ?

bcoca commented:
to_uuid?

gaqzi commented:
`to_uuid` is a lot clearer, and to the point, thanks @bcoca. I've updated the pull request to use `to_uuid` instead. I've also created an Ansible UUID namespace that is now used instead of the DNS namespace.

bcoca commented on docsite/rst/playbooks_variables.rst:
```diff
@@ -409,6 +409,16 @@ To work with Base64 encoded strings::
     {{ encoded | b64decode }}
     {{ decoded | b64encode }}
 
```
this is adding the recently remove sha1 doc

gaqzi replied:
oops, sorry about that, unsuccessful merge rebase on my end. Fixed now!

Also noticed the format for when a filter was introduced is added in the docs, so updated to match that. :)

gaqzi committed:
Add filter to turn a string into a UUID

This filter was made because I needed to create idempotent UUIDs when
installing the agent for Go (http://go.cd), which uses UUIds to
distinguish the agents from each other.

It uses a newly created Ansible namespace to distinguish UUIDs created
by Ansible from any other source. The new namespace is a random one
created by uuidgen on OSX.
